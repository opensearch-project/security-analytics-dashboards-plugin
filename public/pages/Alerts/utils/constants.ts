/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { ALERT_STATE } from '../../../utils/constants';

export const DISABLE_ACKNOWLEDGED_ALERT_HELP_TEXT = 'Only active alerts can be acknowledged.';

export enum AlertSeverity {
  ONE = '1',
  TWO = '2',
  THREE = '3',
  FOUR = '4',
  FIVE = '5',
}

export const severityOptions = [
  { value: 'ALL', text: 'All severity levels' },
  { value: AlertSeverity.ONE, text: '1 (Highest)' },
  { value: AlertSeverity.TWO, text: '2 (High)' },
  { value: AlertSeverity.THREE, text: '3 (Medium)' },
  { value: AlertSeverity.FOUR, text: '4 (Low)' },
  { value: AlertSeverity.FIVE, text: '5 (Lowest)' },
];

export const stateOptions = [
  { value: 'ALL', text: 'All alerts' },
  { value: ALERT_STATE.ACTIVE, text: 'Active' },
  { value: ALERT_STATE.ACKNOWLEDGED, text: 'Acknowledged' },
  { value: ALERT_STATE.COMPLETED, text: 'Completed' },
  { value: ALERT_STATE.ERROR, text: 'Error' },
  { value: ALERT_STATE.DELETED, text: 'Deleted' },
];
