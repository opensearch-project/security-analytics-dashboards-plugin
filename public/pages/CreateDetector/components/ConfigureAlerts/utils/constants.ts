/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { AlertCondition, TriggerAction } from '../../../../../../models/interfaces';

export const MAX_ALERT_CONDITIONS = 10;
export const MIN_ALERT_CONDITIONS = 0;

// SEVERITY_OPTIONS have the id, value, label, and text fields because some EUI components
// (e.g, EuiComboBox) require value/label pairings, while others
// (e.g., EuiCheckboxGroup) require id/text pairings.
export const ALERT_SEVERITY_OPTIONS = {
  HIGHEST: { id: '1', value: '1', label: '1 (Highest)', text: '1 (Highest)' },
  HIGH: { id: '2', value: '2', label: '2 (High)', text: '2 (High)' },
  MEDIUM: { id: '3', value: '3', label: '3 (Medium)', text: '3 (Medium)' },
  LOW: { id: '4', value: '4', label: '4 (Low)', text: '4 (Low)' },
  LOWEST: { id: '5', value: '5', label: '5 (Lowest)', text: '5 (Lowest)' },
};

export const RULE_SEVERITY_OPTIONS = {
  CRITICAL: { id: '1', value: 'critical', label: 'Critical', text: 'critical' },
  HIGH: { id: '2', value: 'high', label: 'High', text: 'High' },
  MEDIUM: { id: '3', value: 'medium', label: 'Medium', text: 'Medium' },
  LOW: { id: '4', value: 'low', label: 'Low', text: 'Low' },
  INFORMATIONAL: { id: '5', value: 'informational', label: 'Info', text: 'Info' },
};

export const EMPTY_DEFAULT_TRIGGER_ACTION: TriggerAction = {
  id: '',
  name: '',
  destination_id: '',
  subject_template: {
    source: '',
    lang: 'mustache',
  },
  message_template: {
    source: '',
    lang: 'mustache',
  },
  throttle_enabled: false,
  throttle: {
    value: 10,
    unit: 'MINUTES',
  },
};

export const EMPTY_DEFAULT_ALERT_CONDITION: AlertCondition = {
  name: '',
  sev_levels: [],
  tags: [],
  actions: [EMPTY_DEFAULT_TRIGGER_ACTION],
  types: [],
  severity: '1',
  ids: [],
};

export const MIN_NUM_NOTIFICATION_CHANNELS = 1;
export const MAX_NUM_NOTIFICATION_CHANNELS = 5;

export const MIN_NUM_RULES = 1;
export const MAX_NUM_RULES = 5;

// Only allows letters. No spaces, numbers, or special characters.
export const MIN_NUM_TAGS = 0;
export const MAX_NUM_TAGS = 5;

export const CHANNEL_TYPES = ['slack', 'email', 'chime', 'webhook', 'ses', 'sns'];
