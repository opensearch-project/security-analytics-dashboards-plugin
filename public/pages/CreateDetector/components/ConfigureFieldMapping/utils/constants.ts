/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  FieldMappingPropertyMap,
  GetFieldMappingViewResponse,
} from '../../../../../../server/models/interfaces';

export const STATUS_ICON_PROPS = {
  unmapped: { type: 'alert', color: 'danger' },
  mapped: { type: 'checkInCircleFilled', color: 'success' },
};

export const EMPTY_FIELD_MAPPINGS_VIEW: GetFieldMappingViewResponse = {
  properties: {},
  unmapped_field_aliases: [],
  unmapped_index_fields: [],
};

export const EMPTY_FIELD_MAPPINGS: FieldMappingPropertyMap = {
  properties: {},
};
