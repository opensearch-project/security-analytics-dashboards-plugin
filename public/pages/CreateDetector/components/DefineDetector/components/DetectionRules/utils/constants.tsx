/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiBasicTableColumn, EuiLink, EuiSwitch } from '@elastic/eui';
import React, { ReactNode } from 'react';
import { RuleItem } from '../types/interfaces';

export type ActiveToggleOnChangeEvent = React.BaseSyntheticEvent<
  React.MouseEvent<HTMLButtonElement>,
  HTMLButtonElement,
  EventTarget & { checked: boolean }
>;

export const getRulesColumns = (
  allEnabled: boolean,
  onAllRulesToggled?: (enabled: boolean) => void,
  onActivationToggle?: (item: RuleItem, active: boolean) => void
): EuiBasicTableColumn<RuleItem>[] => {
  const columns: EuiBasicTableColumn<RuleItem>[] = [
    {
      field: 'name',
      name: 'Rule name',
      render: (ruleName: string, item: RuleItem): ReactNode => (
        <EuiLink style={{ marginLeft: 10 }}>{ruleName}</EuiLink>
      ),
    },
    {
      field: 'severity',
      name: 'Rule severity',
    },
    {
      field: 'logType',
      name: 'Log type',
    },
    {
      field: 'library',
      name: 'Source',
    },
    {
      field: 'description',
      name: 'Description',
    },
  ];

  if (onActivationToggle) {
    columns.unshift({
      name: onAllRulesToggled ? (
        <EuiSwitch
          checked={allEnabled}
          onChange={(event: ActiveToggleOnChangeEvent) => {
            onAllRulesToggled(!allEnabled);
          }}
          label={''}
          showLabel={false}
        />
      ) : undefined,
      render: (item: RuleItem) => {
        return (
          <EuiSwitch
            checked={item.active}
            onChange={(event: ActiveToggleOnChangeEvent) =>
              onActivationToggle(item, event.target.checked)
            }
            label={''}
            showLabel={false}
          />
        );
      },
      width: '60px',
    });
  }

  return columns;
};
