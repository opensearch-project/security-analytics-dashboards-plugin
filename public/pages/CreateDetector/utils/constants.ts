/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DetectorCreationStepInfo } from '../models/interfaces';
import { DetectorCreationStep } from '../models/types';

export const createDetectorSteps: Record<DetectorCreationStep, DetectorCreationStepInfo> = {
  [DetectorCreationStep.DEFINE_DETECTOR]: {
    title: 'Define detector',
    step: 1,
  },
  [DetectorCreationStep.CONFIGURE_FIELD_MAPPING]: {
    title: 'Configure field mapping',
    step: 2,
  },
  [DetectorCreationStep.CONFIGURE_ALERTS]: {
    title: 'Set up alerts',
    step: 3,
  },
  [DetectorCreationStep.REVIEW_CREATE]: {
    title: 'Review and create',
    step: 4,
  },
};
