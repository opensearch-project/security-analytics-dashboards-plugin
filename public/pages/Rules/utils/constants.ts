/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const ruleTypes: string[] = [
  'network',
  'dns',
  'apache_access',
  'windows',
  'ad_ldap',
  'linux',
  'cloudtrail',
  's3',
];

export const ruleSeverity: { name: string; value: string }[] = [
  { name: 'Critical', value: 'critical' },
  { name: 'High', value: 'high' },
  { name: 'Medium', value: 'medium' },
  { name: 'Low', value: 'low' },
  { name: 'Informational', value: 'informational' },
];

export const ruleSource: string[] = ['Sigma', 'Custom'];

export const ruleStatus: string[] = ['Select a rule status', 'experimental', 'test', 'stable'];
