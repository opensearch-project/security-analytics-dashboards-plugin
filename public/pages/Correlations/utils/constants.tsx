/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { CorrelationGraphData } from '../../../../types';
import { ruleSeverity, ruleTypes } from '../../Rules/utils/constants';
import { FilterItem } from '../components/FilterGroup';
import { EuiIcon, EuiTitle } from '@elastic/eui';
import { logTypeCategories, logTypesByCategories } from '../../../utils/constants';
import { getLogTypeLabel } from '../../LogTypes/utils/helpers';

export const graphRenderOptions = {
  nodes: {
    shape: 'dot',
  },
  edges: {
    arrows: {
      to: {
        enabled: false,
      },
    },
    color: '#000000',
    chosen: true,
  },
  layout: {
    hierarchical: false,
    randomSeed: 2222,
    improvedLayout: false,
  },
  autoResize: true,
  height: '800px',
  width: '100%',
  physics: {
    stabilization: {
      fit: true,
      iterations: 1000,
    },
  },
  interaction: {
    zoomView: true,
    zoomSpeed: 0.2,
    dragView: true,
    dragNodes: false,
    multiselect: true,
    tooltipDelay: 50,
    hover: true,
  },
};

export const getDefaultLogTypeFilterItemOptions: () => FilterItem[] = () => {
  const options: FilterItem[] = [];
  logTypeCategories.forEach((category) => {
    const logTypes = logTypesByCategories[category];
    options.push({
      name: (
        <EuiTitle size="xxs">
          <h4>{category}</h4>
        </EuiTitle>
      ),
      id: category,
      checked: 'on',
      childOptionIds: new Set(logTypes.map(({ name }) => name)),
      visible: true,
    });

    logTypes.forEach(({ name }) => {
      options.push({
        name: getLogTypeLabel(name),
        id: name,
        checked: 'on',
        visible: true,
      });
    });
  });

  return options;
};

export const defaultSeverityFilterItemOptions: FilterItem[] = Object.values(ruleSeverity).map(
  (sev) => {
    return {
      name: (
        <p>
          <EuiIcon type={'dot'} color={sev.color.background} /> {`${sev.name}`}
        </p>
      ),
      id: sev.value,
      checked: 'on',
      visible: true,
    };
  }
);

export const emptyGraphData: CorrelationGraphData = {
  graph: {
    nodes: [],
    edges: [],
  },
  events: {},
};

export const getLabelFromLogType = (logType: string) => {
  return ruleTypes.find((ruleType) => ruleType.value === logType)?.label || logType;
};

export const getSeverityLabel = (sev: string) => {
  return ruleSeverity.find((severity) => severity.value === sev)?.name || '';
};

export const getSeverityColor = (sev: string) => {
  return (
    ruleSeverity.find((severity) => severity.value === sev.toLowerCase())?.color || {
      background: 'white',
      text: 'black',
    }
  );
};

export function getNodeSize(severity: string) {
  switch (severity) {
    case 'critical':
      return 40;
    case 'high':
      return 32;
    case 'medium':
      return 24;
    case 'low':
      return 17;
    case 'informational':
    default:
      return 10;
  }
}
