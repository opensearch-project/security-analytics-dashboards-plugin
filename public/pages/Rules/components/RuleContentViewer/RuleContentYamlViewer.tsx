/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiCodeBlock } from '@elastic/eui';
import React from 'react';
import { mapRuleToYamlObject, mapYamlObjectToYamlString } from '../../utils/mappers';
import { Rule } from '../../../../../types';

export interface RuleContentYamlViewerProps {
  rule: Rule;
}

export const RuleContentYamlViewer: React.FC<RuleContentYamlViewerProps> = ({ rule }) => {
  const yamlObject = mapRuleToYamlObject(rule);
  const ruleYaml = mapYamlObjectToYamlString(yamlObject);

  return (
    <EuiCodeBlock language="yaml" data-test-subj={'rule_flyout_yaml_rule'} isCopyable={true}>
      {ruleYaml}
    </EuiCodeBlock>
  );
};
