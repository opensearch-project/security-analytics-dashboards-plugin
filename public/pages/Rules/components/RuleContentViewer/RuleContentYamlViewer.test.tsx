/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import { RuleContentYamlViewer } from './RuleContentYamlViewer';
import { DEFAULT_RULE_UUID } from '../../../../../common/constants';

describe('<RuleContentYamlViewer /> spec', () => {
  it('renders the component', () => {
    const { container } = render(
      <RuleContentYamlViewer
        rule={{
          id: DEFAULT_RULE_UUID,
          category: 'windows',
          title: 'Testing rule',
          description: 'Testing Description',
          status: 'experimental',
          author: 'Bhabesh Raj',
          references: [
            {
              value: 'https://securelist.com/operation-tunnelsnake-and-moriya-rootkit/101831',
            },
          ],
          tags: [
            {
              value: 'attack.persistence',
            },
            {
              value: 'attack.privilege_escalation',
            },
            {
              value: 'attack.t1543.003',
            },
          ],
          log_source: {},
          detection:
            'selection:\n  Provider_Name: Service Control Manager\n  EventID: 7045\n  ServiceName: ZzNetSvc\ncondition: selection\n',
          level: 'high',
          false_positives: [
            {
              value: 'Unknown',
            },
          ],
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
