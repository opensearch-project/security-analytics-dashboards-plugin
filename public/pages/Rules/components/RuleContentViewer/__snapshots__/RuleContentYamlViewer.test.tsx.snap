// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RuleContentYamlViewer /> spec renders the component 1`] = `
<div
  class="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingLarge euiCodeBlock--hasControls prismjs language-yaml"
>
  <pre
    class="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
    tabindex="-1"
  >
    <code
      class="euiCodeBlock__code yaml"
      data-test-subj="rule_flyout_yaml_rule"
    >
      <span
        class="euiCodeBlock__line"
      >
        <span
          class="token key atrule"
        >
          id
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         25b9c01c
        <span
          class="token punctuation"
        >
          -
        </span>
        350d
        <span
          class="token punctuation"
        >
          -
        </span>
        4b95
        <span
          class="token punctuation"
        >
          -
        </span>
        bed1
        <span
          class="token punctuation"
        >
          -
        </span>
        836d04a4f324

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          logsource
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token key atrule"
        >
          product
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         windows

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          title
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         Testing rule

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          description
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         Testing Description

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          tags
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token punctuation"
        >
          -
        </span>
         attack.persistence

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token punctuation"
        >
          -
        </span>
         attack.privilege_escalation

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token punctuation"
        >
          -
        </span>
         attack.t1543.003

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          falsepositives
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token punctuation"
        >
          -
        </span>
         Unknown

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          level
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         high

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          status
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         experimental

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          references
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token punctuation"
        >
          -
        </span>
         https
        <span
          class="token punctuation"
        >
          :
        </span>
        //securelist.com/operation
        <span
          class="token punctuation"
        >
          -
        </span>
        tunnelsnake
        <span
          class="token punctuation"
        >
          -
        </span>
        and
        <span
          class="token punctuation"
        >
          -
        </span>
        moriya
        <span
          class="token punctuation"
        >
          -
        </span>
        rootkit/101831

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          author
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         Bhabesh Raj

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
        <span
          class="token key atrule"
        >
          detection
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token key atrule"
        >
          selection
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        

      </span>
      <span
        class="euiCodeBlock__line"
      >
            
        <span
          class="token key atrule"
        >
          Provider_Name
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         Service Control Manager

      </span>
      <span
        class="euiCodeBlock__line"
      >
            
        <span
          class="token key atrule"
        >
          EventID
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         
        <span
          class="token number"
        >
          7045
        </span>
        

      </span>
      <span
        class="euiCodeBlock__line"
      >
            
        <span
          class="token key atrule"
        >
          ServiceName
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         ZzNetSvc

      </span>
      <span
        class="euiCodeBlock__line"
      >
          
        <span
          class="token key atrule"
        >
          condition
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         selection

      </span>
      <span
        class="euiCodeBlock__line"
      >
        
      </span>
    </code>
  </pre>
  <div
    class="euiCodeBlock__controls"
  >
    <div
      class="euiCodeBlock__copyButton"
    >
      <span
        class="euiToolTipAnchor"
      >
        <button
          aria-label="Copy"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall"
          type="button"
        >
          EuiIconMock
        </button>
      </span>
    </div>
  </div>
</div>
`;
